src/SamplesApp/SamplesApp.UITests.Generator/SamplesListGenerator.cs

```diff
@@ -0,0 +1,270 @@
+							default: return null;
+						}
+					})
+					.ToArray();
+			}
+		}
+
+		private object GetAttributePropertyValue(AttributeData attr, string name)
+			=> attr.NamedArguments.FirstOrDefault(kvp => kvp.Key == name).Value.Value;
+
+		private object GetConstructorParameterValue(AttributeData info, string name)
+			=> info.ConstructorArguments.IsDefaultOrEmpty
+				? default
+				: info.ConstructorArguments.ElementAt(GetParameterIndex(info, name)).Value;
+
+		private int GetParameterIndex(AttributeData info, string name)
+			=> info
+				.AttributeConstructor
+				.Parameters
+				.Select((p, i) => (p, i))
+				.Single(p => p.p.Name == name)
+				.i;
+	
+		private string AlignName(string v)
+			=> v.Replace("/", "_").Replace(" ", "_").Replace("-", "_");
+
+		private void GenerateTests(
+			string assembly,
+			SourceGeneratorContext context,
+			IEnumerable<(INamedTypeSymbol symbol, string[] categories, string name, bool ignoreInSnapshotTests, bool isManual)> symbols)
+		{
+			var groups = 
+				from symbol in symbols.Select((v, i) => (index:i, value:v))
+				group symbol by symbol.index / 50 into g
+				select new {
+					Index = g.Key,
+					Symbols = g.AsEnumerable().Select(v => v.value)
+				};
+
+			foreach(var group in groups)
+			{
+				string sanitizedAssemblyName = assembly.Replace(".", "_");
+				var groupName = $"Generated_{sanitizedAssemblyName}_{group.Index:000}";
+
+				var builder = new IndentedStringBuilder();
+
+				builder.AppendLineInvariant("using System;");
+
+				using (builder.BlockInvariant($"namespace {context.GetProjectInstance().GetPropertyValue("RootNamespace")}.Snap"))
+				{
+					builder.AppendLineInvariant("[global::NUnit.Framework.TestFixture]");
+
+					// Required for https://github.com/unoplatform/uno/issues/1955
+					builder.AppendLineInvariant("[global::SamplesApp.UITests.TestFramework.TestAppModeAttribute(cleanEnvironment: true, platform: Uno.UITest.Helpers.Queries.Platform.iOS)]");
+
+					using (builder.BlockInvariant($"public partial class {groupName} : SampleControlUITestBase"))
+					{
+						foreach (var test in group.Symbols)
+						{
+							builder.AppendLineInvariant("[global::NUnit.Framework.Test]");
+							builder.AppendLineInvariant($"[global::NUnit.Framework.Description(\"runGroup:{group.Index % GroupCount:00}, automated:{test.symbol.ToDisplayString()}\")]");
+
+							if (test.ignoreInSnapshotTests)
+							{
+								builder.AppendLineInvariant("[global::NUnit.Framework.Ignore(\"ignoreInSnapshotTests is set for attribute\")]");
+							}
+							if (test.isManual)
+							{
+								builder.AppendLineInvariant("[global::NUnit.Framework.Ignore(\"isManualTest is set for attribute\")]");
+							}
+
+							builder.AppendLineInvariant("[global::SamplesApp.UITests.TestFramework.AutoRetry]");
+							// Set to 60 seconds to cover possible restart of the device
+							builder.AppendLineInvariant("[global::NUnit.Framework.Timeout(60000)]");
```

Comment:
Interesting - so now there'll be a unit test for every (non-excluded) sample?

Reply:
This part is actually not used :) It's a copy/paste from the other test. I'll remove it.